apps/base/mixins.py

```diff
@@ -68,61 +62,71 @@ def __strip_cookie(cookie_string):
 
         return headers
 
-    def _log_bad_request(self):
+    def _log_bad_request(self, request):
         """Log information about a request if something fails to validate"""
         request_data = {
-            'params': self.request.GET,
-            'data': self.data if self.data else self.request.body,
-            'user_id': self.request.user.id,
-            'headers': self._clean_headers(copy.deepcopy(self.request.META)),
-            'method': self.request.method,
-            'allowed_methods': self.allowed_methods
+            'params': request.GET,
+            'data': request.data if request.data else request.body,
+            'user_id': request.user.id,
+            'headers': self._clean_headers(copy.deepcopy(request.META)),
+            'method': request.method,
         }
 
         logger.warning(
-            'Invalid {} data supplied to {}'.format(self.request.method, self.__class__.__name__),
+            'Invalid {} data supplied to {}'.format(request.method, self.__class__.__name__),
             extra=request_data
         )
 
-    def _handle_bad_request(self, request, *args, **kwargs):
-        """
-        Handles creating an instance of `base.responses.BadResponse` to return the request.
-        Follows mostly the same format as the Django implementation of returning a response.
-        :param request: (Request) WSGI request object
-        :return: (BadRequest) Instance of response indicating a bad request
-        """
-        self._log_bad_request()
-        response = BadRequest('Invalid {} data supplied to {}'.format(request.method, self.__class__.__name__))
-        self.headers = self.default_response_headers
-        return self.finalize_response(request, response, *args, **kwargs)
-
-    def dispatch(self, request, *args, **kwargs):
-        # Need to covert Django WSGI request to DRF request in order to access request.data
-        _request = self.initialize_request(request, *args, **kwargs)
-
-        if not hasattr(self, _request.method.lower()):
-            self._log_bad_request()
-            return HttpResponseNotAllowed(self.allowed_methods)
-
-        serializer_class = getattr(self, '{}_request_serializer'.format(_request.method.lower()), None)
+    def _validate_request(self, request):
+        serializer_class = getattr(self, '{}_request_serializer'.format(request.method.lower()), None)
 
         if serializer_class:
-            data = getattr(_request, self.REQUEST_DATA_MAPPING[_request.method], {})
+            data = getattr(request, self.REQUEST_DATA_MAPPING[request.method], {})
             self.data = data
             serializer = serializer_class(data=data)
 
             if serializer.is_valid():
```

Comment:
We can just return `serializer.is_valid()` and let the request method deal with logging the bad request. Here we're just duplicating the return value of the method anyway, might as well simplify it

Reply:
DISREGARD: If we return `serializer.is_valid()` we lose the cleaned data from the serializer, which we need in the views to do processing